grails-app/services/org/broadinstitute/orsp/IssueService.groovy

```diff
@@ -210,12 +211,34 @@ class IssueService {
         input.collect { element ->
             if (issue.getProperties().get(element.key) != null && element.key != "fundings") {
                 updatedIssue.(element.key) = element.value
+                updatedIssue.setUpdateDate(new Date())
             }
         }
         updatedIssue.save(flush:true)
         updatedIssue
     }
 
+    void projectApproval(Issue issue) {
+        Boolean projectGeneralDataApproved = false
+        Boolean projectDocumentsApproved = false
+
+        if (issue != null && !issue.getAttachments().isEmpty()) {
+            projectGeneralDataApproved = issue.getNonEmptyExtraProperties().find {
+                it.name == IssueExtraProperty.PROJECT_REVIEW_APPROVED
+            }?.value
+
+            projectDocumentsApproved = issue.getAttachments().find {
+                it.status != IssueStatus.Approved.getName()
+            } != null
```

Comment:
Just want to understand this logic better - maybe a comment explaining what this is doing would be nice. In this code, we look for any attachment that **IS NOT** approved .... and if we find any, we are setting `projectDocumentsApproved` to true. 

That seems counter-intuitive to me. I would expect, just looking at variable names, that you would set the value to true if and only if **ALL** attachments were approved.

Reply:
Please check again if the new changes resolve your comment.